style/search.js

```diff
@@ -0,0 +1,203 @@
+//Re-map silly results from photon
+function mapResultToDescription(type, key, value) {
+  switch (type) {
+    case "house":
+      switch (key) {
+        case "highway":
+          switch (value) {
+            case "bus_stop":
+              return value;
+            default:
+              return "road";
+          }
+        case "aeroway":
+          switch (value) {
+            case "aerodrome":
+              return "airport";
+            default:
+              return value;
+          }
+        case "railway":
+          return `train ${value}`;
+        case "man_made":
+        case "tourism":
+        case "amenity":
+        case "leisure":
+          return value;
+        case "office":
+        default:
+          return key;
+      }
+    case "district":
+    case "locality":
+      switch (key) {
+        case "landuse":
+          return `${value} area`;
+        default:
+          return value;
+      }
+    case "tunnel":
+      return key;
+  }
+  return type;
+}
+
+function geocoderResultEntry(result) {
+  let p = result.properties;
+  console.log(p);
+  let type = mapResultToDescription(p.type, p.osm_key, p.osm_value).replaceAll(
+    "_",
+    " "
+  );
+
+  let addr = collapseArray([p.housenumber, p.street], " ");
+  let name = p.name;
+
+  if (name == undefined) {
+    name = addr;
+    addr = undefined;
+  }
+
+  let description = collapseArray(
+    [addr, p.city, p.county, p.state, p.country],
+    ", "
+  );
+
+  let item = document.createElement("div");
+  item.className = "gc-result-item";
+
+  let itemCategory = document.createElement("div");
+  itemCategory.className = "gc-result-category";
+  itemCategory.innerText = type;
+  let itemName = document.createElement("div");
+  itemName.className = "gc-result-name";
+  itemName.innerText = name;
+  let itemDescription = document.createElement("div");
+  itemDescription.className = "gc-result-description";
+  itemDescription.innerText = description;
+
+  item.appendChild(itemCategory);
+  item.appendChild(itemName);
+  item.appendChild(itemDescription);
+
+  return item;
+}
+
+function geocoderResponse(data) {
+  liveResults.innerHTML = "";
+
+  for (let i = 0; i < data.features.length; i++) {
+    resultGeometry[i] = data.features[i].properties.extent;
+    resultPoint[i] = data.features[i].geometry.coordinates;
+    let result = geocoderResultEntry(data.features[i]);
+    liveResults.appendChild(result);
+    result.onclick = (e) => goToResult(i);
+  }
+}
+
+function search(e) {
+  let queryTerm = e.target.value;
+  resultSelectIndex = -1;
+  if (queryTerm.length < 3) {
+    liveResults.innerHTML = "";
+    return;
+  }
+
+  let position = map.getCenter();
+  let searchQuery =
+    "https://photon.komoot.io/api/?q=" +
+    encodeURIComponent(e.target.value) +
+    "&lat=" +
+    position.lat +
+    "&lon=" +
+    position.lng +
+    "&limit=3";
+
+  fetch(searchQuery)
+    .then((response) => response.json())
+    .then((data) => geocoderResponse(data));
```

Comment:
I'd suggest writing this with async/await, but it also needs updated to cancel all but the most recent request.